src/main/java/io/strimzi/kproxy/codec/Frame.java

```diff
@@ -22,6 +22,8 @@
  * A frame in the Kafka protocol, which may or may not be fully decoded.
  */
 public interface Frame {
+
+    int estimateEncodeSize();
```

Comment:
Rename `estimateEncodeSize` → `estimateEncodedSize`

Also, this needs some doc, in particular when and in what ways is this estimate allowed to be wrong?